interface/modules/custom_modules/oe-module-ehi-exporter/table.sql

```diff
@@ -0,0 +1,65 @@
+#used for cleaning up the tables if needed
+#DROP TABLE IF EXISTS ehi_export_job_task_patients;
+#DROP table IF EXISTS ehi_export_job_tasks;
+#DROP TABLE IF EXISTS ehi_export_job_patients;
+#DROP TABLE IF EXISTS ehi_export_job;
+
+#IfNotTable
+CREATE TABLE `ehi_export_job`(
+   `ehi_export_job_id` int(11) NOT NULL AUTO_INCREMENT,
+   `uuid` BINARY(16) DEFAULT NULL,
+   `user_id` BIGINT(20) NOT NULL COMMENT 'FK to users.id - represents the user that started the export process',
+   `creation_date` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+   `completion_date` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+   `status` VARCHAR(20) NOT NULL DEFAULT 'processing' COMMENT 'processing=export in progress, failed=error occurred in one or more tasks, completed=export completed without errors',
+   `include_patient_documents` TINYINT(1) NOT NULL DEFAULT 1 COMMENT '1=Include patient documents into zip export, 0=no patient documents included in the export',
+   `document_limit_size` INT(11) UNSIGNED NOT NULL DEFAULT 524288000 COMMENT 'The batch zip file export size, default set to 500MB max value size is 4GB',
+   PRIMARY KEY(`ehi_export_job_id`),
+   CONSTRAINT `FK_ehi_export_job_users` FOREIGN KEY (`user_id`) REFERENCES `users`(`id`) ON DELETE RESTRICT ON UPDATE RESTRICT
```

Comment:
is there any way to avoid foreign keys? We have tried foreign keys in past and as I recall it caused lots of breaks when collation/encoding issues are not aligned (which are beyond OpenEMR's control) in addition to some other issues. At that time, goal was to avoid foreign keys like the plague :)

Reply:
There shouldn't be any collation / encoding issues as all the foreign keys are against numeric columns.  If there are other issues I'm interested in hearing them.  I could definitely see problems if the FKs were against the list_options tables or other text data types.  With numeric columns I don't see the problem.  I'd rather keep the foreign keys as it keeps data integrity, but this isn't a really critical table so its not required if you insist we avoid it.

If you feel strongly I can drop them but I'd rather not as they are intended to maintain referential integrity which is something OpenEMR cold benefit from.  The FKs are there in order to break queries if the queries corrupt the database.  This whole project would have been a lot easier if we actually had FKs defined on most of our numeric columns such as the patient pid.